src/Microsoft.Crank.AzureDevOpsWorker/DevopsMessage.cs

```diff
@@ -191,49 +157,41 @@ public async Task<bool> UpdateTaskTimelineRecordAsync(string taskLogObject)
                 count = 1
             };
 
+            return SendAsync(updateTimelineUrl, body);
+        }
+
+        private async Task<bool> SendAsync<TBody>(
+            string url, TBody body, [CallerMemberName] string callerName = "")
+        {
             var requestBody = JsonSerializer.Serialize(body);
 
             try
             {
-                var result = await PatchDataAsync(updateTimelineUrl, requestBody);
+                var result = await PostDataAsync(url, requestBody);
                 return result.IsSuccessStatusCode;
             }
-            catch (Exception e)
+            catch (Exception ex)
             {
-                Console.WriteLine($"UpdateTaskTimelineRecordAsync failed: {updateTimelineUrl}");
-                Console.WriteLine(e.ToString());
+                Console.WriteLine($"{callerName} failed: {url}");
+                Console.WriteLine(ex);
                 return false;
             }
         }
 
-        
-        private async Task<HttpResponseMessage> PostDataAsync(string url, HttpContent content)
-        {
-            var response = await _httpClient.PostAsync(new Uri(url), content);
-
-            return response;
-        }
+        private Task<HttpResponseMessage> PostDataAsync(string url, HttpContent content) =>
+            _httpClient.PostAsync(new Uri(url), content);
 
         private async Task<HttpResponseMessage> PostDataAsync(string url, string requestBody)
         {
             var buffer = Encoding.UTF8.GetBytes(requestBody);
-            var byteContent = new ByteArrayContent(buffer);
+            using var byteContent = new ByteArrayContent(buffer);
             byteContent.Headers.ContentType = new MediaTypeHeaderValue("application/json");
 
             var response = await _httpClient.PostAsync(new Uri(url), byteContent);
 
             return response;
         }
 
-        private async Task<HttpResponseMessage> PatchDataAsync(string url, string requestBody)
-        {
-            var buffer = Encoding.UTF8.GetBytes(requestBody);
-            var byteContent = new ByteArrayContent(buffer);
-            byteContent.Headers.ContentType = new MediaTypeHeaderValue("application/json");
-
-            var response = await _httpClient.PatchAsync(new Uri(url), byteContent);
-
-            return response;
-        }
+        public void Dispose() => _httpClient?.Dispose();
```

Comment:
The `_httpClient` instance is static, it's reused across messages during the whole app lifetime. This can't be disposed.

Reply:
That is a good callout. I only saw it being used `Program` but see it's called multiple times on the event callback - missed that.